src/prover/validator.rs

```diff
@@ -0,0 +1,32 @@
+use anyhow::anyhow;
+
+use crate::config::Config;
+use crate::prover::identity::Identity;
+
+pub struct ProofValidator {
```

Comment:
Not sure I'm seeing the need for this abstraction.

This is equivalent to a function call with 2 args (depth and id commitments).

Reply:
Especially since IdentityProcessor impls have a copy of `Config` so they can access tree depth no problem

Reply:
Removed. :)